ztp/gitops-subscriptions/argocd/SNOExpansion.md

```diff
@@ -0,0 +1,483 @@
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Writing image to disk
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Waiting for control plane
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Waiting for control plane
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Rebooting
+14fd821b-a35d-9cba-7978-00ddf535ff37   cnfdf15   true       worker   Done
+
+```
+#### __ManagedClusterInfo__
+
+When the worker node installation completes, its certificates are approved automatically. At his point the worker will appear in ManagedClusterInfo status:
+
+```bash
+$ oc get managedclusterinfo/cnfdf15 -n cnfdf15 -o jsonpath='{range .status.nodeList[*]}{.name}{"\t"}{.conditions}{"\t"}{.labels}{"\n"}{end}'
+cnfdf15	[{"status":"True","type":"Ready"}]	{"node-role.kubernetes.io/master":"","node-role.kubernetes.io/worker":""}
+cnfdf16	[{"status":"True","type":"Ready"}]	{"node-role.kubernetes.io/worker":""}
+
+```
+
+### Applying the DU profile
+This procedure assumes the SNO cluster being expanded is provisioned with DU profile, as described in the [README.md](README.md)
+The DU profile is provisioned using `PolicyGenTemplate` resources partitioned into common, group and site-specific manifests.
+For example, for SNO, the git repository linked to the `policies` ArgoCD application will include:
+- [common-ranGen.yaml](example/policygentemplates/common-ranGen.yaml). This template usually contains a set of operator subscriptions and it's unlikely that it should be modified for a worker addition.
+- [group-du-sno-ranGen.yaml](example/policygentemplates/group-du-sno-ranGen.yaml)
+- [example-sno-site.yaml](example/policygentemplates/example-sno-site.yaml)
+- [ns.yaml](example/policygentemplates/ns.yaml)
+- [kustomization.yaml](example/policygentemplates/kustomization.yaml)
+
+The procedure of configuring the DU profile on the worker node is considered as an upgrade. To initiate the upgrade flow, user must update the existing policies, or create an additional ones, and then create a ClusterGroupUpgrade to reconcile the policies in the group of clusters.
+
+#### __Preparations (optional - for ZTP DU profile version 4.11 or below)__
```

Comment:
or we create another manifest for worker mcp

Reply:
This wouldn't help if the SNO was already provisioned with OperatorConfig setting the daemonSelector labels to `master`.  In this case changing the operatorConfig policy to 'worker' would only add `worker` label to the node selector, but it wouldn't remove the `master` label. The worker would remain unselected.
For the future I think we need to change the way we do policies for OperatorConfig - either stop populating daemonset node selector, or set it fixed to `worker`. 
https://github.com/openshift-kni/cnf-features-deploy/pull/1214
https://github.com/openshift-kni/cnf-features-deploy/pull/1213

Reply:
It's tough on SNO, since the `worker` pool is present but doesn't affect the node at all.